src/architecture/system_model/_UnitTest/test_PySysModel.py

```diff
@@ -81,6 +84,46 @@ def test_PySysModel():
 
     assert testResults < 1, testMessage
 
+def test_ErrorPySysModel():
+    """This method tests that exceptions happening in Python module
+    Reset and UpdateState are always printed to sys.stderr"""
+
+    testMessage = []
+
+    mod = ErroringPythonModule()
+
+    simulated_syserr_reset = io.StringIO("")    
+
+    with contextlib.redirect_stderr(simulated_syserr_reset):
+        try:
+            mod.Reset()
+        except Exception:
+            pass
+
+    error_reset = simulated_syserr_reset.getvalue()
+
+    if len(error_reset) == 0:
+        testMessage.append("Reset did not print it's exception")
+    elif not error_reset.rstrip().endswith("ValueError: Error in Reset"):
+        testMessage.append("Reset did not print the correct exception")
+
+    simulated_syserr_update = io.StringIO("")
+
+    with contextlib.redirect_stderr(simulated_syserr_update):
+        try:
+            mod.UpdateState(0)
+        except Exception:
+            pass
+
+    error_update = simulated_syserr_update.getvalue()
+    
+    if len(error_update) == 0:
+        testMessage.append("Reset did not print it's exception")
+    elif not error_update.rstrip().endswith("ValueError: Error in UpdateState"):
+        testMessage.append("UpdateState did not print the correct exception")
+
+    assert len(testMessage) == 0, testMessage
```

Comment:
Can we substitute checking for no error messages with using ``numpy.testing``? You can find more information here: https://numpy.org/doc/stable/reference/routines.testing.html. It's more robust and includes a lot of really useful functions. I'd suggest giving it a try.

Reply:
Happy to be moving to a more robust testing library, but how would you change the checks in here? The first:
```
if len(error_reset) == 0:
        testMessage.append("Reset did not print its exception")
```
could be changed to
```
npt.assert_string_equal(error_reset, "")
```
but the following
```
if not error_reset.rstrip().endswith("ValueError: Error in Reset"):
        testMessage.append("Reset did not print the correct exception")
```
doesn't really have `numpy.testing`  equivalent, right?

Reply:
I think you're right. Now that I think about it, it's overkill for what you're doing here. I'm happy if you keep it the way it is!